src/modules/workspaces/workspaces.controller.ts

```diff
@@ -84,6 +88,37 @@ export class WorkspacesController {
     return this.workspaceUseCases.acceptWorkspaceInvite(user, inviteId);
   }
 
+  @Get('/invitations/:inviteId/validate')
+  @ApiOperation({
+    summary: 'Validates if invitation is valid',
+  })
+  @ApiOkResponse({
+    description: 'Workspace invitation is valid',
+  })
+  async validateWorkspaceInvitation(
+    @Param('inviteId', ValidateUUIDPipe)
+    inviteId: WorkspaceInviteAttributes['id'],
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    try {
+      return await this.workspaceUseCases.validateWorkspaceInvite(inviteId);
+    } catch (error) {
+      let errorMessage = error.message;
+      if (error instanceof BadRequestException) {
+        throw error;
+      } else {
+        Logger.error(
+          `[WORKSPACES/INVITATIONS/:INVITEID/VALIDATE] Error while trying to validate invite with id ${inviteId}, message: ${error}, stack: ${
+            error.stack || 'No stack trace'
+          }`,
+        );
+        res.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        errorMessage = 'Internal server error';
+      }
+      return { message: errorMessage };
+    }
+  }
+
```

Comment:
I know this type of structure is in another endpoints as well, but it is not needed. Do not use the res to return here, just use the exceptions nestjs provides and throw new internalServerException(), we do not need to write any message or something like that

Reply:
Preferably, remove the try-catch and just let the request fail, nestjs will do the work. We are going to add an exceptionFilter logger in order for these kind of things get logged correctly, do not worry about logging in this case (unless it is a REALLY special case, we can let the exceptionFilter handle the logging). Also, without the try-catch remember to remove the return await as we do not need to await there anymore to catch the error